jablib/src/main/java/org/jabref/model/entry/LinkedFile.java

```diff
@@ -225,16 +225,69 @@ public boolean isOnlineLink() {
         return isOnlineLink(link.get());
     }
 
+    /**
+     * Extracts the file name, including basename and extension, from the link.
+     *
+     * @return extracted file name
+     */
+    public String getFileName() {
+        String linkedName = link.get();
+        if (isOnlineLink(linkedName)) {
+            return FileUtil.getFileNameFromUrl(linkedName);
+        } else {
+            try {
+                Path pathname = Path.of(linkedName).getFileName();
+                if (pathname != null) {
+                    return pathname.toString();
+                } else {
+                    return "";
+                }
+            } catch (InvalidPathException ex) {
+                return "";
+            }
+        }
+    }
+
+    public Optional<String> getURI(BibDatabaseContext databaseContext, FilePreferences filePreferences) {
+        List<Path> dirs = databaseContext.getFileDirectories(filePreferences);
+        return getURI(dirs);
+    }
+
+    /**
+     * Tries to associate the file with a URI.
+     * In case of an online link, uses the referenced URL directly.
+     * In case of a file link, searches for the file's path, which may not exist.
+     *
+     * @return String of the URI that points to the file
+     */
+    public Optional<String> getURI(List<Path> directories) {
```

Comment:
It's an URL, because it is a **location**, not an **identifier**.

Please rename it.